### PR TITLE
Major update of build Artemis broker image

### PR DESCRIPTION
**Pull request by lenosi**
## Major update of build Artemis broker image

- Based on UBI 7 minimal
- Use java-1.8.0-openjdk-headless
- Use normalized production PATHs
- Use a secure custom application user
- Add EXPOSED ports
- Add Volumes [etc, data, tmp]

- Removed unused cct module
- Provide Artemis init, install, post modules
- Update entry.sh as an entry point

## Image build size

```
2020-09-08 14:36:12,300 cekit        INFO     Original image size: 513.71 MB
2020-09-08 14:36:12,301 cekit        INFO     Squashed image size: 426.40 MB
```

## Other features
- Added pre-commit hooks to ensure lint and coding standards are followed.
- yamllint
- Replace AsciiDoc with Markdown
- Add PR Github action with pre-commit

### CODE REVIEW

**Comment by lenosi** [comment]
I need to mention one potential regression, it's about default values for a container.

```yaml
  - name: "AMQ_USER"
    value: "amq_user"
  - name: "AMQ_PASSWORD"
    value: "amq_password"
```

The reason is to have an image that runs from the default values.
Running with simply `docker run artemiscloud/activemq-artemis-broker`

File: image.yaml [hunk 1]
```diff
@@ -1,56 +1,77 @@
...
 schema_version: 1
 
 name: "artemiscloud/activemq-artemis-broker"
 description: "ActiveMQ Artemis broker container image"
 version: "2.13.0"
+<<<<<<< HEAD
 from: "ubi7:7.8-345"
+=======
+from: "registry.access.redhat.com/ubi8/ubi-minimal:latest"
+>>>>>>> a8e7ebc... Major update of build Artemis broker image
```
Comment (gaohoward) [inline image.yaml]: Can you resolve this?

Comment (lenosi) [reply image.yaml]: It was reverted for ubi7.

**Comment by brusdev** [comment]
This PR has been closed because it has not been updated in the last 6 months. Feel free to reopen it if it is still valid.

**Pull request closed without merging**